### ISSUE TITLE
Same IDs can appear in train and test set when using bootstrap for outer resampling in nested setup

### ISSUE THREAD

**Issue by sebffischer** [mlr-org/mlr3#1369]
When running a nested resampling and using the `rsmp("bootstrap")` for the outer resampling, the same original observations can appear in the train and test set of the inner resampling.

There might be scenarios where one wants this, but there might also be cases where one wants to have all observations that have the same `row_id` (there can be more than one when using bootstrap) appear either in the train **or** test set of the inner resampling. This is demonstrated in the code below:

``` r
library(mlr3verse)
#> Loading required package: mlr3
lgr::get_logger("mlr3")$set_threshold("warn")
set.seed(1L)
at = auto_tuner(
  learner = lrn("classif.debug", x = to_tune(), predict_sets = c("train", "test")),
  resampling = rsmp("holdout"),
  measure = msr("classif.ce"),
  tuner = tnr("random_search"),
  term_evals = 1,
  store_benchmark_result = TRUE
)

rr = resample(tsk("iris"), at, rsmp("bootstrap", repeats = 1), store_models = TRUE)
bmr = rr$learners[[1L]]$model$tuning_instance$archive$benchmark_result
rr = bmr$resample_result(1)
test_ids = rr$prediction("test")$row_ids
train_ids = rr$prediction("train")$row_ids
any(test_ids %in% train_ids)
#> [1] TRUE
```

One possible solution would be to have an argument `row_ids_as_groups` in the parameter set of the `Resampling`. I.e. before `$instantiate()`ing a `Resampling` on a task, we would set the row role "groups" to treat all observations with the same `row_id` as one group.

**Comment by sebffischer** [comment]
decision: not a bug, don’t change this

**Comment by bblodfon** [comment]
This is actually a good argument to not use bootstrap on the outer resampling of a nested resampling scheme.

**Comment by bblodfon** [comment]
For future reference, I randomly stumpled upon something similar in [Hothorn et al. (2005)](https://www.tandfonline.com/doi/abs/10.1198/106186005X59630), where the authors remove the duplicate observations from the inner validation sets:

<img width="1104" height="326" alt="Image" src="https://github.com/user-attachments/assets/af9ae41f-ab30-4426-82d2-0efb1e59e994" />
